types/node/ts3.9/index.d.ts

```diff
@@ -0,0 +1,8 @@
+// NOTE: These definitions support NodeJS and TypeScript 3.5 - 3.6.
+// This is required to enable typing assert in ts3.7 without causing errors
+// Typically type modifications should be made in base.d.ts instead of here
+
+/// <reference path="base.d.ts" />
+
+/// <reference path="../ts3.4/assert.d.ts" />
+/// <reference path="../ts3.4/util.promisify.d.ts" />
```

Comment:
This file and directory is unnecessary, since `util.promisify.d.ts` doesn’t differ between `ts3.4` and `ts3.9`.

Reply:
Is there a way to limit the signature to TS 4.0+ without it?

Reply:
And I thought the ts3.6 directory exists because TS 3.7 added some new features that are being used. This should be the same - new feature starting with TS 4.0?

Reply:
I think the point @AlCalzone is trying to make here is that `ts3.4/util.promiseify.d.ts` isn't included in the set of referenced files (it is referenced by `ts3.6/index.d.ts` but not by `ts3.6/base.d.ts`).

Reply:
Honestly, I'm not sure what to do here. 
There are some existing definitions that should be active for TS >= 3.7 (thus the 3.6 directory). 
Then there are the new promisify signatures which should be active for TS >= 4.0.
And there are the old promisify signatures which need to be available for all versions <= 3.9.

I got it to work the way it is now, but if this is not the correct way, I'm happy to change it.